### PR TITLE
Adds Tests to Abort AI Functions when No Empire is Available

### PR DESCRIPTION
**Pull request by geoffthemedio**
I noticed this branch sitting without a pull request. I don't remember why I pushed it, but it was probably due to a forum thread or issue. I can't find which that was, though. Regardless, the safety checks added here might be useful, so I thought I'd rebase and make a new pull request.

### CODE REVIEW

File: default/python/AI/FreeOrionAI.py [hunk 1]
```diff
@@ -202,6 +226,16 @@ def generateOrders():  # pylint: disable=invalid-name
     at end of this function, fo.doneTurn() should be called to indicate to the client that orders are finished
     and can be sent to the server for processing."""
     empire = fo.getEmpire()
+    if empire is None:
+        print "This client has no empire. Doing nothing to generate orders."
+        try:
+            # early abort if no empire. no need to do meter calculations
+            # on last-seen gamestate if nothing can be ordered anyway...
+            fo.doneTurn()
+        except Exception as e:
```
Comment (LGM-Doyle) [inline default/python/AI/FreeOrionAI.py]: This looks odd, but is correct.  I'd add a comment that `doneTurn()` is issued on behalf of the client network id, not the empire id, so not having a correct empire id does not invalidate `doneTurn()`

Comment (geoffthemedio) [reply default/python/AI/FreeOrionAI.py]: comment added

**Comment by geoffthemedio** [comment]
rebased. futher comments / concerns?

**Comment by LGM-Doyle** [comment]
LGTM

**Pull request merged**